src/blocked/Blocked.tsx

```diff
@@ -0,0 +1,58 @@
+import React, { Component } from "react";
+import { Button, Header, Modal } from "semantic-ui-react";
+import { Runtime } from "webextension-polyfill-ts";
+
+interface Props {
+  sendMessage: (
+    message: { type: string; params: unknown },
+    options?: Runtime.SendMessageOptionsType | undefined
+  ) => Promise<unknown>;
+}
+
+export class Blocked extends Component<Props> {
+  readonly url: string;
+
+  constructor(props: Props) {
+    super(props);
+    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+    this.url = new URLSearchParams(window.location.search).get("url")!;
+    this.acknowledgeRisk = this.acknowledgeRisk.bind(this);
+  }
+
+  async acknowledgeRisk(): Promise<void> {
+    await this.props.sendMessage({
+      type: "resetQuality",

```

Comment:
Should the quality be reset? Shouldn't it be set to the new quality?

Reply:
Since we don't have the whole context in this page and therefore not the new quality, this is not possible. It is sufficient to reset it to unknown since the user gets redirected to the website he initially wanted to access on the button click, which sets the new quality.